src/NuGet.Clients/NuGet.Credentials/PluginCredentialProviderBuilder.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.Configuration;
+
+namespace NuGet.Credentials
+{
+    /// <summary>
+    /// Discovers configured plugin providers.
+    /// </summary>
+    public class PluginCredentialProviderBuilder
+    {
+        private readonly Configuration.ISettings _settings;
+        private readonly Configuration.IEnvironmentVariableReader _envarReader;
+
+        public PluginCredentialProviderBuilder(Configuration.ISettings settings) : this(settings, new EnvironmentVariableWrapper())
+        {
+        }
+
+        public PluginCredentialProviderBuilder(Configuration.ISettings settings, Configuration.IEnvironmentVariableReader envarReader)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+            if (envarReader == null)
+            {
+                throw new ArgumentNullException(nameof(envarReader));
+            }
+
+            _settings = settings;
+            _envarReader = envarReader;
+
+        }
+
+        /// <summary>
+        /// Find all configured plugin providers.
+        /// Plugin providers are entered in settings files with a key prefixed
+        /// "CredentialProvider.Plugin." and with a value that is an absolute or
+        /// relative path to a plugin application.  Relative paths will be searched from
+        /// the following roots:  
+        /// 1. Current executing assembly directory
```

Comment:
In NuGet.Config, a relative path is relative to the nuget.config file, not relative to the executing assembly.